python/lsst/sims/catUtils/utils/LightCurveGenerator.py

```diff
@@ -0,0 +1,496 @@
+
+        for iobs, obs in enumerate(obs_list):
+            group_dex = -1
+
+            for ix, obs_g in enumerate(obs_groups):
+                dd = haversine(obs._pointingRA, obs._pointingDec,
+                               obs_list[obs_g[0]]._pointingRA, obs_list[obs_g[0]]._pointingDec)
+                if dd < tol:
+                    group_dex = ix
+                    break
+
+            if group_dex == -1:
+                obs_groups.append([iobs])
+                mjd_groups.append([obs_list[iobs].mjd.TAI])
+            else:
+                obs_groups[group_dex].append(iobs)
+                mjd_groups[group_dex].append(obs_list[iobs].mjd.TAI)
+
+        # rearrange each group of ObservationMetaDatas so that they
+        # appear in chronological order by MJD
+        obs_groups_out = []
+        for ix, (grp, mjd) in enumerate(zip(obs_groups, mjd_groups)):
+            oo = np.array(grp)
+            mm = np.array(mjd)
+            dexes = np.argsort(mm)
+            obs_groups_out.append([obs_list[ii] for ii in oo[dexes]])
+
+        return obs_groups_out
+
+    def _get_query_from_group(self, grp, chunk_size):
+        """
+        Take a group of ObervationMetaData that all point to the same region
+        of the sky.  Query the CatSim database for all of the celestial objects
+        in that region, and return it as an iterator over database rows.
+        """
+
+        cat = self._lightCurveCatalogClass(self._catalogdb, obs_metadata=grp[0])
+
+        cat.db_required_columns()
+
+        query_result = cat.db_obj.query_columns(colnames=cat._active_columns,
+                                                obs_metadata=cat.obs_metadata,
+                                                constraint=cat.constraint,
+                                                chunk_size=chunk_size)
+
+        return query_result
+
+    def _light_curves_from_query(self, cat_dict, query_result, grp):
+        """
+        Read in an iterator over database rows and return light curves for
+        all of the objects contained.
+
+        Input parameters:
+        -----------------
+        cat_dict is a dict of InstanceCatalogs keyed on bandpass name.  There
+        only needs to be one InstanceCatalog per bandpass name.  These dummy
+        catalogs provide the methods needed to calculate synthetic photometry.
+
+        query_result is an iterator over database rows that correspond to
+        celestial objects in our field of view.
+
+        grp is a list of ObservationMetaData objects that all point to the
+        region of sky containing the objects in query_result.  cat_dict
+        should contain an InstanceCatalog for each bandpass represented in
+        grp.
+
+        Output
+        ------
+        This method does not output anything.  It adds light curves to the
+        instance member variables self.mjd_dict, self.mag_dict, self.sig_dict,
+        and self.truth_dict.
+        """
+
+        global _sed_cache
```

Comment:
If local_gamma_cache is a cache storing gamma values for each pointing group (fieldID), it might be good to clarify this in a comment.